packages/hebao_v1/contracts/modules/transfers/TransferModule.sol

```diff
@@ -58,38 +58,51 @@ abstract contract TransferModule is SecurityModule
         )
         internal
     {
-        if (amount == 0) return;
-
         if (token == address(0)) {
             transactCall(wallet, to, amount, "");
         } else {
-            bytes memory txData = abi.encodeWithSelector(
-                ERC20(0).transfer.selector,
-                to,
-                amount
+            require(
+                transactTokenTransfer(wallet, token, to, amount),
+                "TRANSFER_FAILED"
             );
-            transactCall(wallet, token, 0, txData);
         }
         emit Transfered(wallet, token, to, amount, logdata);
     }
 
     function approveInternal(
         address wallet,
         address token,
-        address to,
+        address spender,
         uint    amount
         )
         internal
+        returns (uint additionalAllowance)
     {
         require(token != address(0), "UNSUPPORTED");
 
-        bytes memory txData = abi.encodeWithSelector(
-            ERC20(0).approve.selector,
-            to,
-            amount
+        // Current allowance
+        uint allowance = ERC20(token).allowance(wallet, spender);
+
+        // First reset the approved amount if needed
+        bytes memory txData;
+        if (allowance > 0) {
```

Comment:
Shall we skip calling transactTokenApprove if the current allowance is already >= amount?